### ISSUE TITLE
RPC calls in toolbox-app need some attention

### ISSUE THREAD

**Issue by Stefie** [polkadot-js/apps#1281]
I noticed while working on the `child_storage`, that the RPC call section in the toolbox-app could need some attention. Some calls are not working properly or as expected and optional parameters are represented as required fields in the ui.

The Bug I found is:
https://polkadot.js.org/apps/#/toolbox -> RPC ->
`state` > `getStorage() || getStorageHash() || getStorageKey()`

Then use for example `0x50a63a871aced22e88ee6466fe5aa5d9` (`api.query.sudo.key.key()`) or `0xf4039aa8ae697861be900c58239e96f7` `api.query.balances.creationFee.key()` as `key: StorageKey`, both are just random examples.

The results in the following error:
<img width="1257" alt="Screenshot 2019-06-06 at 11 36 36" src="https://user-images.githubusercontent.com/125398/59023135-66342c80-884f-11e9-96d7-ac4f7ec0ec93.png">

The call is working outside of `apps` (in my boilerplate setup with `apiPromise()`) and when you're calling it directly with curl, so I'd guess that the Bug is neither in `substrate` nor `api`.

I would guess that this could be a similar blackbox as https://github.com/polkadot-js/apps/issues/1160 where one bug turned into several.

It would make sense to:
- Check all methods. 
- Turn optional parameters into optional fields in the UI

**Comment by jacogr** [comment]
Marking as enhancement and P5. The Optional has never been taken care of. 

Since I only have so many hours in a day to code and I know of 1 user in the wild, would rather lean to removing it completely if this is causing issues to be logged.

**Comment by Stefie** [comment]
If the choice is between removing it or keeping it slightly buggy I'd probably vote for keeping it. So far it was only me and @Tbaut stumbling into issues. We're both working full-time with polkadot/substrate so there seem to be internal use-cases for having it.

I actually like debugging and would have had a look myself, but I just expect that it could be rather time consuming, that's why I didn't touch it today. If it's fine with you just leave the ticket open, I might come back to it later.

**Comment by Stefie** [comment]
This seems to spread like a virus and needs some attention after all.  
I'm getting these rpc-core errors on every block update for `subscribeNewHead()`, but also in other places all over Apps.
`Bytes: required length less than remainder, expected at least xxx, found xx`

**Comment by jacogr** [comment]
You will get those -

- ensure you are running at least substrate with this commit - https://github.com/paritytech/substrate/commit/3c1c8da5357cfad708d20e0d807a94821029f4cb (if you want to be able to send anything)
- ensure that you are least running substrate with this commit - https://github.com/paritytech/substrate/commit/e4dd6d0db862e2f8fcee5dfb6950ac9d7b0d435f (if you want to be able to decode events)
- ensure that you are running at least apps with this commit https://github.com/polkadot-js/apps/commit/28c5a8380963716699e5022f70a285ad102b2f5a (to enable all the new types)

Basically - if you are seeing decoding errors, you are not on the latest

**Comment by Stefie** [comment]
Ah, great, I didn't see Demis changes from yesterday yet. All working smoothly with the later substrate master again.

**Comment by jacogr** [comment]
It was all Basti's fixes.

**Comment by jacogr** [comment]
Closing. Optionals are supported in the UI for a while now. Not going to dig around for specific issues, if there are specific ones that are problematic, please do log.
